### ISSUE TITLE
Jackett index timeout causes full timeout

### ISSUE THREAD

**Issue by stryan** [lardbit/nefarious#119]
This is more of an FYI then a full Issue report, but I figured I'd let people know in case it is a full issue.

If one of the indexers in Jackett takes too long (over 60 seconds) to respond, it will result in all querys to Jackett timing out. This occurs even if there are other Indexers available and/or the misbehaving indexer is unneeded for the query.

This is difficult to notice, as Nefarious will only report a timeout error contacting the Jacket server entirely. Validating the Indexers under settings doesn't show the error, as the page will lock up until the timeout occurs with no error messages, and the settings section will look fine since the other parts respond correctly.

**Comment by lardbit** [comment]
You're right, and this is due to the fact nefarious is taking advantage of jackett's *aggregate indexer* feature https://github.com/Jackett/Jackett#aggregate-indexers which has the drawback of the whole request fails if a single indexer fails. I chose the aggregate option out of convenience but maybe it's time to reconsider and just query each indexer individually and then comb throttle results afterwards to find the best match.

I'll keep this issue open until I have time to work on it. For the time being, the settings page should issue an error, though, so I'll try and fix that first.

Thanks for pointing this out. It's bit me a few times too and I had to remove the failing indexer to resolve.

**Comment by lardbit** [comment]
For the meantime, I fixed the **Settings** page to at least display an error of which Jackett indexer failed.

![image](https://user-images.githubusercontent.com/45122868/120242394-642cba80-c22a-11eb-9490-7368e6984f5a.png)

**Comment by stryan** [comment]
Great! Honestly, that's all I needed.
